homeassistant/components/jewish_calendar/config_flow.py

```diff
@@ -0,0 +1,153 @@
+    DEFAULT_DIASPORA,
+    DEFAULT_HAVDALAH_OFFSET_MINUTES,
+    DEFAULT_LANGUAGE,
+    DEFAULT_NAME,
+    DOMAIN,
+)
+
+LANGUAGE = [
+    SelectOptionDict(value="hebrew", label="Hebrew"),
+    SelectOptionDict(value="english", label="English"),
+]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, str]:
+    """Validate the user input."""
+    return {"title": data[CONF_NAME]}
+
+
+class JewishCalendarConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Jewish calendar."""
+
+    VERSION = 1
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry: ConfigEntry) -> OptionsFlow:
+        """Get the options flow for this handler."""
+        return JewishCalendarOptionsFlowHandler(config_entry)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors = {}
+
+        if user_input is not None:
+            for entry in self.hass.config_entries.async_entries(DOMAIN):
+                if all(
+                    _value == entry.as_dict().get(_key)
+                    for _key, _value in user_input.items()
+                    if _key != CONF_NAME
+                ):
+                    return self.async_abort(reason="already_configured")
+
+                if entry.data[CONF_NAME] == user_input[CONF_NAME]:
+                    errors[CONF_NAME] = "name_exists"
+                    break
+
+            try:
+                info = await validate_input(self.hass, user_input)
+                return self.async_create_entry(title=info["title"], data=user_input)
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(CONF_NAME, default=DEFAULT_NAME): str,
+                    vol.Required(
+                        CONF_DIASPORA, default=DEFAULT_DIASPORA
+                    ): BooleanSelector(),
+                    vol.Required(
+                        CONF_LANGUAGE, default=DEFAULT_LANGUAGE
+                    ): SelectSelector(SelectSelectorConfig(options=LANGUAGE)),
+                    vol.Required(
+                        CONF_LOCATION,
+                        default={
+                            "latitude": self.hass.config.latitude,
+                            "longitude": self.hass.config.longitude,
+                        },
+                    ): LocationSelector(),
+                    vol.Optional(
+                        CONF_ELEVATION, default=self.hass.config.elevation
+                    ): int,
+                    vol.Optional(
+                        CONF_TIME_ZONE, default=self.hass.config.time_zone
+                    ): str,
```

Comment:
Can we make it a dropdown like this?
```suggestion
                    ): SelectSelector(
                        TextSelectorConfig(
                            options=sorted(list(zoneinfo.available_timezones())),
                        ),
```

Reply:
Can't find this use anywhere, it's seems to work though